SwiftCompilerSources/Sources/SIL/Type.swift

```diff
@@ -214,6 +215,16 @@ public struct Type : TypeProperties, CustomStringConvertible, NoReflectionChildr
     }
     return false
   }
+
+  public func getEnumCasePayload(caseIdx: Int, function: Function) -> Type {
```

Comment:
You don't need this. Use `getEnumCases` instead. If you like you can add a utility function in `EnumCases` to find the payload type of a specific index. This should have an optional return `Type?`, though.